crates/bevy_asset/src/asset_server.rs

```diff
@@ -213,16 +213,15 @@ impl AssetServer {
     }
 
     pub fn load<'a, T: Asset, P: Into<AssetPath<'a>>>(&self, path: P) -> Handle<T> {
-        self.load_untyped(path).typed()
+        self.load_untyped(path.into()).typed()
     }
 
     // TODO: properly set failed LoadState in all failure cases
-    async fn load_async<'a, P: Into<AssetPath<'a>>>(
+    async fn load_async<'a>(
```

Comment:
For now I think this is reasonable because we don't export it. But some potential future `pub async load_async()` variant should be generic to have parity with load().